test/basic.js

```diff
@@ -98,6 +98,16 @@ require('tap').test('basic', function (t) {
       raw: 'git+ssh://git@notgithub.com/user/foo#1.2.3'
     },
 
+    'git+ssh://git@notgithub.com:user/foo#1.2.3': {
```

Comment:
We should also have a `'git+ssh://git@notgithub.com:8080/file/dir#1.2.3'` case that parses as a URL.